### PR TITLE
Add headers for requesting and receiving the server version

### PR DESCRIPTION
**Pull request by josephschorr**

### CODE REVIEW

File: pkg/requestmeta/requestmeta.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+const (
+	// RequestServerVersion, if specified in a request header, asks SpiceDB to return its
+	// server version in the response header (if supported).
+	// Value: `1`
+	RequestServerVersion BoolRequestMetadataHeaderKey = "io.spicedb.requestversion"
+)
+
+// AddRequestHeaders returns a new context with the given values as request headers.
+func AddRequestHeaders(ctx context.Context, keys ...BoolRequestMetadataHeaderKey) context.Context {
+	values := map[RequestMetadataHeaderKey]string{}
```
Comment (jzelinskie) [inline pkg/requestmeta/requestmeta.go:24]:
```go
values := make(map[RequestMetadataHeaderKey]string, len(keys))
```

File: pkg/requestmeta/requestmeta.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+const (
+	// RequestServerVersion, if specified in a request header, asks SpiceDB to return its
+	// server version in the response header (if supported).
+	// Value: `1`
+	RequestServerVersion BoolRequestMetadataHeaderKey = "io.spicedb.requestversion"
+)
+
+// AddRequestHeaders returns a new context with the given values as request headers.
+func AddRequestHeaders(ctx context.Context, keys ...BoolRequestMetadataHeaderKey) context.Context {
```
Comment (jzelinskie) [inline pkg/requestmeta/requestmeta.go:23]: Shouldn't this be `AddBooleanRequestHeaders`? I imagine there will be headers in the future that have contents.

**Review by jzelinskie — CHANGES_REQUESTED** [review]

File: pkg/requestmeta/requestmeta.go (hunk 2 above)
Comment (josephschorr) [reply pkg/requestmeta/requestmeta.go:23]: The idea is that you can Add boolean headers, but Set non-bool headers (since they have values)

Comment (jzelinskie) [reply pkg/requestmeta/requestmeta.go:23]: ah, yeah that makes sense

**Review by jzelinskie — APPROVED** [review]

**Review by jzelinskie — APPROVED** [review]

**Pull request merged**